pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -364,7 +357,9 @@ protected Message<T> internalReceive() throws PulsarClientException {
             message = incomingMessages.take();
             decreaseIncomingMessageSize(message);
             checkState(message instanceof TopicMessageImpl);
-            unAckedMessageTracker.add(message.getMessageId(), message.getRedeliveryCount());
+            if (listener == null) {
+                unAckedMessageTracker.add(message.getMessageId(), message.getRedeliveryCount());
+            }
```

Comment:
seem the listener only use `nternalReceive(long timeout, TimeUnit unit)`, change here seems to be unnecessary